clique-grpc/src/lib.rs

```diff
@@ -1,18 +1,171 @@
+use futures::{
+    stream::futures_unordered::FuturesUnordered,
+    sync::{mpsc, oneshot},
+    Async, Future, Poll, Stream,
+};
+use http::Uri;
+use hyper::client::connect::{Destination, HttpConnector};
+use tokio;
+use tower_hyper::{
+    body::Body,
+    client::{self, Connection},
+    util,
+};
+use tower_request_modifier;
+use tower_service::Service;
+use tower_util::MakeService;
+
+use std::collections::HashMap;
+
 pub mod clique_proto {
     include!(concat!(env!("OUT_DIR"), "/messaging.rs"));
 }
 
-pub use self::clique_proto::*;
+use clique_proto::client::{self as ProtoClient, MembershipService};
+use clique_proto::RapidRequest;
+
+struct PoolReq {
+    uri: Uri,
+    req: RapidRequest,
+}
+
+struct FuturesRunner<Request, T: Service<Request>> {
+    inner: FuturesUnordered<Box<dyn Future<Item = T::Response, Error = ()>>>,
+    rx: mpsc::UnboundedReceiver<(Request, oneshot::Sender<T::Response>)>,
+}
+
+impl<Request, T> FuturesRunner<Request, T>
+where
+    T: Service<Request> + Clone,
+    Request: Address,
+{
+    pub fn new() -> mpsc::UnboundedSender<(Request, oneshot::Sender<T::Response>)> {
+        let (tx, rx) = mpsc::unbounded();
+        tokio::spawn(FuturesRunner {
+            inner: FuturesUnordered::new(),
+            rx,
+        });
+
+        tx
+    }
 
-use clique::messaging::{Broadcast, RapidResponseFuture};
+    fn poll_enqueue(&mut self) {
+        self.rx
+            .for_each(|uri| {
+                let dst = Destination::try_from_uri(uri.clone()).unwrap();
+                let connector = util::Connector::new(HttpConnector::new(4));
+                let settings = client::Builder::new().http2_only(true).clone();
+                let mut make_client = client::Connect::with_builder(connector, settings);
 
-struct CliqueClient;
+                let client = make_client
+                    .make_service(dst) // we need to enqueue
+                    .map_err(|e| panic!("Unable to connect to destination"));
 
-impl Broadcast for CliqueClient {
-    fn broadcast(&self, req: RapidRequest) -> RapidResponseFuture {
-        unimplemented!()
+                self.inner.push(Box::new(client));
+                Ok(())
+            })
+            .poll();
     }
-    fn set_membership(&self, reciepients: Vec<Endpoint>) {
-        unimplemented!()
+}
+
+impl<Request, T> Future for FuturesRunner<Request, T>
+where
+    T: Service<Request> + Clone,
+    Request: Address,
+{
+    type Item = T::Response;
```

Comment:
if this is a background task you prob want it to be `(), ()` since that is the requirement for spawning a future. It should do all the return values over a channel.